src/services/websocket.rs

```diff
@@ -21,7 +21,7 @@ cfg_if! {
 }
 
 /// A status of a websocket connection. Used for status notification.
-#[derive(Debug, PartialEq)]
+#[derive(Clone, Copy, Debug, PartialEq)]
```

Comment:
Do you need `Copy`? I would prefer leaving it out because we may put a `String` in `Error` at some point.

Reply:
I mistakenly thought it was however after removing it from my branch and testing I see that it isn't.  My apologies and thanks for catching that.

Reply:
No worries! Looks good now, thanks!